src/native-stack/views/NativeStackView.tsx

```diff
@@ -73,7 +78,7 @@ const MaybeNestedStack = ({
 
   const isHeaderInModal = isAndroid
     ? false
-    : stackPresentation !== 'push' && headerShown === true;
+    : stackPresentation !== 'push' && headerShown;
```

Comment:
```suggestion
    : stackPresentation !== 'push' && headerShown === true;
```

It's JavaScript, let's be explicit.